### DISCUSSION TITLE
Rivers and channels

### DISCUSSION THREAD

**Discussion by mymyrere** [noaa-ocs-modeling/OCSMesh#148] [General]
Hi,
I would like to define channel/rivers in my script.
I have used this tutorial:
https://schism-dev.github.io/schism/master/mesh-generation/meshing-for-compound-floods/generate-river-map.html
to create my shapefiles and outputs the shapefile to use for OCSmesh.
I am just wondering how do you use those shapefile,
- Do you use the polygon and use "add_region_constraint" fucntion
- Do you use the arcs? if yes which function do you use with that shapefile, do you put nodes along those lines? 

I know this discussion has been done before but I can't find any exemple.

Thanks

**Comment by SorooshMani-NOAA** [comment]
The effort to integrate the rivers into the mesh is an ongoing one. I worked on it in the past, but kind of gave up after some failed attempts. @felicio93 is currently working on it.

In other words, if you're only interested in setting a small mesh size function where those rivers are, you can use the thalwegs extracted by the rivermapper tool and add it as a feature (`hfun.add_feature`); but if you're looking for something to integrate those polygons as it is with the mesh, then that's what @felicio93  is working on
